src/utils.ts

```diff
@@ -0,0 +1,27 @@
+import { SchemaAnalyzer, SchemaParseOptions } from './schema-analyzer';
+import { AnyIterable } from './types';
+
+export function verifyStreamSource(
+  source: AnyIterable
+): AnyIterable {
+  if (!(Symbol.iterator in source) && !(Symbol.asyncIterator in source)) {
+    throw new Error(
+      'Unknown input type for `docs`. Must be an array, ' +
+        'stream or MongoDB Cursor.'
+    );
+  }
+
+  return source;
+}
+
+export async function getCompletedSchemaAnalyzer(
```

Comment:
nit: I like to avoid adding functions to `util` files when possible as I feel they can sometimes absorb a lot of the code and overtime that can make discovery more difficult. Would it make sense to move these functions to the `schema-analyzer.ts` file? No changes needed, mostly a suggestion. Also I should mention I like not having much code in `index.ts` files so I'm happy these are at least not there anymore!